topology/probes/ovn/ovn.go

```diff
@@ -0,0 +1,309 @@
+func (l *switchPortLinker) GetBALinks(lpNode *graph.Node) (edges []*graph.Edge) {
+	probe := l.probe
+	uuid, _ := lpNode.GetFieldString("UUID")
+	for _, ls := range l.probe.ovndbapi.GetLogicSwitches() {
+		for _, lp := range l.probe.ovndbapi.GetLogicPortsBySwitch(ls.Name) {
+			if lp.UUID == uuid {
+				if lsNode, _ := probe.lsIndexer.GetNode(ls.UUID); lsNode != nil {
+					link, err := topology.NewLink(probe.graph, lsNode, lpNode, topology.OwnershipLink, nil)
+					if err != nil {
+						logging.GetLogger().Error(link)
+						continue
+					}
+					edges = append(edges, link)
+				}
+			}
+		}
+	}
+	return edges
+}
+
+type aclLinker struct {
+	probe *Probe
+}
+
+// GetABLinks returns all the links from a specified port group to its ACLs
+func (l *aclLinker) GetABLinks(lsNode *graph.Node) (edges []*graph.Edge) {
+	name, _ := lsNode.GetFieldString("Name")
+	for _, acl := range l.probe.ovndbapi.GetACLsBySwitch(name) {
+		if aclNode, _ := l.probe.aclIndexer.GetNode(acl.UUID); aclNode != nil {
+			if link, _ := topology.NewLink(l.probe.graph, lsNode, aclNode, topology.OwnershipLink, nil); link != nil {
+				edges = append(edges, link)
+			}
+		}
+	}
+	return edges
+}
+
+// GetBALinks returns all the links from a port group to the specified ACL
+func (l *aclLinker) GetBALinks(aclNode *graph.Node) (edges []*graph.Edge) {
+	uuid, _ := aclNode.GetFieldString("UUID")
+	for _, ls := range l.probe.ovndbapi.GetLogicSwitches() {
+		for _, acl := range l.probe.ovndbapi.GetACLsBySwitch(ls.Name) {
+			if acl.UUID == uuid {
+				if lsNode, _ := l.probe.lsIndexer.GetNode(ls.UUID); lsNode != nil {
+					link, err := topology.NewLink(l.probe.graph, lsNode, aclNode, topology.OwnershipLink, nil)
+					if err != nil {
+						logging.GetLogger().Error(link)
+						continue
+					}
+					edges = append(edges, link)
+				}
+			}
+		}
+	}
+	return edges
+}
+
+func (p *Probe) registerNode(indexer *graph.Indexer, uuid string, metadata graph.Metadata) {
+	logging.GetLogger().Debugf("Registering OVN object with UUID %s and metadata %+v", uuid, metadata)
+
+	p.graph.Lock()
+	defer p.graph.Unlock()
+
+	id := graph.GenID(uuid)
+	node, _ := indexer.GetNode(uuid)
+	if node == nil {
+		n, err := p.graph.NewNode(id, metadata)
+		if err != nil {
+			logging.GetLogger().Error(err)
+			return
+		}
+		node = n
+	} else {
+		p.graph.SetMetadata(node, metadata)
+	}
+	indexer.Index(node, map[string]interface{}{uuid: uuid})
+}
+
+func (p *Probe) unregisterNode(indexer *graph.Indexer, uuid string) {
+	logging.GetLogger().Debugf("Unregistering OVN object with UUID %s", uuid)
+
+	p.graph.Lock()
+	defer p.graph.Unlock()
+
+	node, _ := indexer.GetNode(uuid)
+	if node != nil {
+		p.graph.DelNode(node)
+		indexer.Unindex(node)
+	}
+}
+
+func (p *Probe) logicalSwitchMetadata(ls *goovn.LogicalSwitch) graph.Metadata {
+	m := graph.Metadata{
+		"Type":    "logical_switch",
+		"Name":    ls.Name,
+		"Manager": "ovn",
+		"UUID":    ls.UUID,
+	}
+	for k, v := range ls.ExternalID {
```

Comment:
For OVS we put all the external keys in a subsection EXtID. It may be nice to keep the same behavior

Reply:
Done